app/models/repository.rb

```diff
@@ -29,12 +29,17 @@
 
 class Repository < ActiveRecord::Base
   include Redmine::Ciphering
+  include OpenProject::Scm::ManageableRepository
 
   belongs_to :project
   has_many :changesets, order: "#{Changeset.table_name}.committed_on DESC, #{Changeset.table_name}.id DESC"
 
   before_save :sanitize_urls
 
+  # Managed repository lifetime
+  after_save :create_managed_repository, if: Proc.new { |repo| repo.managed? }
+  after_destroy :delete_managed_repository, if: Proc.new { |repo| repo.managed? }
```

Comment:
Ok, I get your point.

What I feel is that even now, you don't really have the full transaction. What you have covered in a transaction is:
* Creating repository in DB
* Creating job in DB
The actual creation on the FS is not covered by the transaction. So we will need some form of cleanup in any case.

I think you already pointed out the need for error handling somewhere, but I can't seem to find it.

Currently, as far as I have seen it, there is no error handling on the delayed job. E.g. dequeuing the delayed job at some point and notifying the user somehow, that he should wait no longer as it will never happen.

And whatever the strategy for this is, this will also reflect the cleanup we have to do. 

So I don't think that we gain much from the transaction as it does not cover all of what really needs to be done.